theseus/geometry/so3.py

```diff
@@ -52,15 +54,16 @@ def __str__(self) -> str:
             return f"SO3(matrix={self.data}), name={self.name})"
 
     def _adjoint_impl(self) -> torch.Tensor:
-        return self.data
+        return self.data.clone()
```

Comment:
I was thinking whether this was necessary when I did the first past to the previous version of this PR, but I wasn't sure. Did you find that not cloning was giving errors (maybe in the test?).

Reply:
I added clone() mainly because of encapsulation. I'm a little concerned that users might have access to and modify self.data through adjoint() without clone().

Of course, I can remove clone() and there will be no errors.

Reply:
This is reasonable. Let's keep as is.